File: app/src/main/java/com/rdapps/uncheckableradiobutton/UncheckableRadioButton.java
package com.rdapps.uncheckableradiobutton;

import android.content.Context;
import android.content.res.TypedArray;
import android.support.v7.widget.AppCompatRadioButton;
import android.util.AttributeSet;
import android.widget.RadioGroup;

public class UncheckableRadioButton extends AppCompatRadioButton {

    private boolean uncheckable = false;

    public UncheckableRadioButton(Context context) {
        super(context);
    }

    public UncheckableRadioButton(Context context, AttributeSet attrs) {
        super(context, attrs);
        init(context, attrs);
    }

    public UncheckableRadioButton(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        init(context, attrs);
    }

    private void init(Context context, AttributeSet attrs) {
        TypedArray typedArray = context.obtainStyledAttributes(attrs, R.styleable.UncheckableRadioButton);
        uncheckable = typedArray.getBoolean(R.styleable.UncheckableRadioButton_uncheckable, false);
        typedArray.recycle();
    }

    @Override
    public void toggle() {

        if (uncheckable) {
            if (isChecked()) {
                if (getParent() != null && getParent() instanceof RadioGroup) {
                    ((RadioGroup) getParent()).clearCheck();
                }
            } else {
                super.toggle();
            }
        } else {
            super.toggle();
        }
    }

    @Override
    public CharSequence getAccessibilityClassName() {
        return UncheckableRadioButton.class.getName();
    }
}

File: app/src/main/java/com/rdapps/uncheckableradiobutton/UncheckableRadioButton.kt
package com.rdapps.uncheckableradiobutton

import android.content.Context
import android.util.AttributeSet
import android.widget.RadioGroup
import androidx.appcompat.widget.AppCompatRadioButton

class UncheckableRadioButton @JvmOverloads constructor(
    context: Context, attrs: AttributeSet? = null
) : AppCompatRadioButton(context, attrs) {
    private var uncheckable = false

    init {
        init(context, attrs)
    }

    private fun init(context: Context, attrs: AttributeSet?) {
        val typedArray = context.obtainStyledAttributes(attrs, R.styleable.UncheckableRadioButton)
        uncheckable = typedArray.getBoolean(R.styleable.UncheckableRadioButton_uncheckable, false)
        typedArray.recycle()
    }

    override fun toggle() {
        if (uncheckable) {
            if (isChecked) {
                if (parent != null && parent is RadioGroup) {
                    (parent as RadioGroup).clearCheck()
                }
            } else {
                super.toggle()
            }
        } else {
            super.toggle()
        }
    }

    override fun getAccessibilityClassName(): CharSequence {
        return UncheckableRadioButton::class.java.name
    }
}